### ISSUE TITLE
Templ Language Support (Golang)

### ISSUE THREAD

**Issue by prnk28** [zed-industries/zed#7351]
### Check for existing issues

- [X] Completed

### Language

Templ

### Tree Sitter parser link

https://github.com/vrischmann/tree-sitter-templ/tree/master

### Language server link

https://marketplace.visualstudio.com/items?itemName=a-h.templ

**Comment by tego101** [comment]
I just set the language to GO for now and it does somewhat of a good job.

**Comment by joerdav** [comment]
Awesome, this might work somewhat for highlighting but the autocomplete and diagnostics capability will require the plugin.

**Comment by wtfiscrq** [comment]
if no one's working on this I can give it a shot with the new Zed extensions

**Comment by SomeoneToIgnore** [comment]
Do note though, that current Zed extensions only support syntax highlights, but not the LSP-related features.

**Comment by wtfiscrq** [comment]
thanks for the heads-up @SomeoneToIgnore, are any plans to support LSPs through extensions for the future that you're aware of?

anyway having syntax highlighting would already be useful. I keep having to select Go whenever I open new files, which is a bit tedious.

**Comment by SomeoneToIgnore** [comment]
Yes, the work is ongoing now as far as I am aware, but that might take time since currently it's closer to prototyping phase.
